ipfs/library/Network/IPFS/Files.hs

```diff
@@ -0,0 +1,155 @@
+module Network.IPFS.Files (cp, rm, statCID, write) where
+
+import           Data.ByteString.Lazy.Char8  as CL
+
+import           Network.IPFS.Client.Files   as Files
+import           Network.IPFS.File.Types
+import           Network.IPFS.Files.Error    as IPFS.Files
+import qualified Network.IPFS.Internal.UTF8  as UTF8
+import           Network.IPFS.Remote.Class   as IPFS
+import           Network.IPFS.Prelude        hiding (link)
+import qualified Network.IPFS.Process.Error  as Process
+import           Network.IPFS.CID.Types      as IPFS ( mkCID, CID(CID) )
+import           Network.IPFS.Types          as IPFS
+
+import qualified Network.IPFS.Client.Files.Statistics.Types as Files.Statistics
+import qualified Network.IPFS.Client.Files.Write.Form.Types as Write
+
+import qualified RIO.ByteString.Lazy         as Lazy
+import qualified RIO.Text                    as Text
+
+import           Servant.Client
+import           Servant.Multipart           as Multipart
+import qualified Servant.Multipart.Client    as Multipart.Client
+
+
+{-| MFS Copying.
+
+Can be used for both copying to and from MFS.
+Will create parent directories.
+
+-}
+cp :: (MonadLogger m, MonadRemoteIPFS m)
+  => Either IPFS.CID FilePath
+  -> FilePath
+  -> m (Either IPFS.Files.Error ())
+cp cidOrFilePath destination =
+  let
+    from =
+      case cidOrFilePath of
+        Left (IPFS.CID hash) -> Text.unpack (Text.append "/ipfs/" hash)
+        Right path           -> path
+
+    args = MfsCopyArgs
+        { from = Text.pack from
+        , to = Text.pack destination
+        , parents = True
+        }
+  in
+  IPFS.mfsCopy args >>= \case
+    Right () ->
+      return $ Right ()
+
+    Left err -> do
+      formattedError <- parseClientError err
+      return <| Left formattedError
```

Comment:
No need to change, but just an FYI we're moving away from the `<|` operator since it's equivalent to `$` (which is used just above on line 51)